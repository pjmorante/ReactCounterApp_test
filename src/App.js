import React, { useState } from "react";
import PropTypes from 'prop-types';

function App( { numInicial } ) {

  const [ counter, setCounter ] = useState( numInicial );

  const handleAdd = () => {
    setCounter(counter + 1)
  }

  const handleSubst = () => setCounter(counter - 1)

  const handleReset = () => setCounter( 0 );

  return (
    <div>
      <h1>{ counter }</h1>
      <button onClick={ handleAdd }>+</button>
      <button onClick={ handleReset }>Reset</button>
      <button onClick={ handleSubst }>-</button>
    </div>
  );
}

App.prototype = {
  value: PropTypes.number.isRequired
}

export default App;
