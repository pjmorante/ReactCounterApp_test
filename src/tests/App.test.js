import { shallow } from 'enzyme';
import React from 'react';
import App from '../App';


describe('Tests on <App />', () => {

    let wrapper;

    beforeEach( () => {

        wrapper = shallow( <App numInicial={10}/> );

    })

    test('should show default value', () => { 

        const counterText = wrapper.find('h1').text().trim();

        expect( counterText ).toBe('10');
    })

     test('should add one with the plus button (+)', () => {

        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h1').text().trim();

        expect( counterText ).toBe('11');
    })

     test('should substrct one with the minus button (-)', () => {   

        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h1').text().trim();

        expect( counterText ).toBe('9');
    })

     test('should reset to zero with the reset button (Reset)', () => {        

        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h1').text().trim();

        expect( counterText ).toBe('0');
    })

     
})