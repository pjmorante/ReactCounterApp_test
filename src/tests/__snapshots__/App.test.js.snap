// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <App /> It must show <App /> 1`] = `
<div>
  <h1 />
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;

exports[`Tests on <App /> It should show <App /> 1`] = `
<div>
  <h1 />
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
